src/main/java/software/amazon/documentdb/jdbc/metadata/DocumentDbSchema.java

```diff
@@ -133,7 +133,9 @@ public void setGetTableFunction(
             @NonNull final Function<String, DocumentDbSchemaTable> getTableFunction,
             @NonNull final Function<Set<String>, Map<String, DocumentDbSchemaTable>> getRemainingTablesFunction)
             throws IllegalStateException {
-        if (this.tables != null || this.tableReferences == null) {
+        if (this.tables != null) {
```

Comment:
@normanj-bitquill Could you explain this further? I'm not sure I understand the need or impact of this change.

Reply:
In my implementation, it is relatively cheap to get the tables. As a result, I am instantiating the DocumentDbSchema object with tables. It does not need to look the tables up on demand later.

The code here was failing because tables was already populated and tableReferences was null.

Reply:
Probably a short comment along the lines...
```suggestion
        // Schema tables already initialized - nothing to do.
        if (this.tables != null) {
```

Reply:
Added a comment.